socketshark/utils.py

```diff
@@ -25,20 +27,48 @@ def _get_rate_limit_wait(log, resp, opts):
             if 0 <= new_wait <= max_wait:
                 wait = new_wait
             elif new_wait > max_wait:
-                log.warn('rate reset value too high',
-                         name=header_name, value=header_value)
+                log.warn(
+                    'rate reset value too high',
+                    name=header_name,
+                    value=header_value,
+                )
                 wait = max_wait
             else:
-                log.warn('invalid rate reset value',
-                         name=header_name, value=header_value)
+                log.warn(
+                    'invalid rate reset value',
+                    name=header_name,
+                    value=header_value,
+                )
         except ValueError:
-            log.warn('invalid rate reset value',
-                     name=header_name, value=header_value)
+            log.warn(
+                'invalid rate reset value',
+                name=header_name,
+                value=header_value,
+            )
     return wait
 
 
+def _scrub_url(url):
+    """Scrub URL username and password."""
+    url_parts = urlsplit(url)
+    if url_parts.password is None:
+        return url
+    else:
+        # url_parts tuple doesn't include password in _fields
+        # so can't easily use _replace to get rid of password
+        # and then call urlunsplit to reconstruct url.
+        if url_parts.port is None:
```

Comment:
You can leave the comment above, but I would reassemble it like this (it was fun to study the implementation):
```
_, _, hostinfo = url_parts.netloc.rpartition('@')
scrubbed_netloc = f'*****:*****@{hostinfo}'
scrubbed_url_parts._replace(netloc=scrubbed_netloc)
return urlunsplit(scrubbed_url)
```

Also, an inconsistency I noticed is that the code wouldn't scrub a username without a password, but if it's a username with a password (which may be empty), it would scrub both. Not sure if this is intended or whether it matters at all.